Control coverage/Feature/SynapseWorkspace.md

```diff
@@ -0,0 +1,175 @@
+
+### Azure Policies or REST APIs used for evaluation 
+
+- REST API to list diagnostic setting details of Synapse workspace resources: {resourceId}/providers/microsoft.insights/diagnosticSettings?api-version=2021-05-01-preview <br />
+**Properties:**
+name<br />
+properties.logs.category<br />
+properties.logs.enabled<br />
+properties.logs.retentionPolicy.enabled<br />
+properties.logs.retentionPolicy.days<br />
+properties.workspaceId<br />
+properties.storageAccountId<br />
+properties.eventHubName<br />
+<br />
+<br />
+
+
+## Azure_Synapse_NetSec_Dont_Allow_Universal_IP_Range 
+
+### Display Name 
+Do not use Any-to-Any IP range for Azure Synapse Workspace
+
+### Rationale 
+Using the firewall feature ensures that access to the data or the service is restricted to a specific set/group of clients. For effective usage, allow only the required IPs. Allowing larger ranges like 0.0.0.0/0, 0.0.0.0/1, 128.0.0.0/1, etc. will defeat the purpose.
+
+### Control Settings 
+```json 
+  {
+    "IPRangeStartIP": "0.0.0.0",
+    "IPRangeEndIP": "255.255.255.255"
+  }
+```
+
+### Control Spec 
+
+> **Passed:**
+>Custom firewall rule (PublicNetworkAccess) is disabled.
+>Custom firewall rule with Any-to-Any IP range with Start IP address as 0.0.0.0 and End Ip address as 255.255.255.255 is NOT found.
+>Firewall rule that gets added from "Allow Azure resources to access this resource" is allowed.
+
+> **Failed:**
+> Custom firewall rule with Any-to-Any IP range with Start IP address as 0.0.0.0 and End Ip address as 255.255.255.255 is found.
+
+
+### Recommendation 
+
+- **Azure Portal** 
+To remediate unmanaged Synapse Workspace remove Any to Any firewall IP address. 
+1. Go to Azure Portal --> your Synapse Workspace  --> Settings --> Networking --> Firewall rules --> Select Any to Any firewall rule (allowAll rule) --> Delete --> Save. 
+For accessing Unmanaged Synapse Workspace add individual firewall IP address. 
+
+To remediate managed Synapse Workspace disable public network access. 
+1. Go to Azure Portal --> your Synapse Workspace  --> Settings --> Networking --> Public network access to workspace endpoints --> Disabled --> Save. For accessing Managed Synapse Workspace, configure private endpoint connection. To configure private endpoint connection refer: https://learn.microsoft.com/en-us/azure/synapse-analytics/security/how-to-connect-to-workspace-with-private-links.
+      
+
+### Azure Policies or REST APIs used for evaluation 
+
+- REST API used to list Synapse workspaces at Subscription level: <br />
+/subscriptions/{subscriptionId}/providers/Microsoft.Synapse/workspaces?api-version=2021-06-01<br />
+
+- REST API used to list Firewall rules for Synapse workspace:
+/subscriptions/{subscriptionId}/resourceGroups/{1}/providers/Microsoft.Synapse/workspaces/{synapseWorkspaceName}/firewallRules?api-version=2021-06-01
+
+**Properties:**
```

Comment:
Please make these properties aligned with the above section.

Reply:
updated